src/logics/Logic.cc

```diff
@@ -1593,3 +1591,20 @@ char* Logic::printTerm        (PTRef tr, bool l, bool s) const { return printTer
 void Logic::termSort(vec<PTRef>& v) const { sort(v, LessThan_PTRef()); }
 
 void  Logic::purify     (PTRef r, PTRef& p, lbool& sgn) const {p = r; sgn = l_True; while (isNot(p)) { sgn = sgn^1; p = getPterm(p)[0]; }}
+
+bool Logic::typeCheck(SymRef sym, vec<PTRef> const & args) const {
+    Symbol const & symbol = sym_store[sym];
+    if (symbol.chainable()) {
+
+    }
+    if (symbol.left_assoc()) {
+
+    }
+    if (symbol.nargs() == args.size_()) {
+        for (unsigned int i = 0; i < symbol.nargs(); i++) {
+            if (symbol[i] != getSortRef(args[i]))
+                return false;
+        }
+    }
+    return true;
```

Comment:
This does not seem right.
I don't get the meaning of the two empty `if`s and for the last one, if the number of arguments do not match, it should definitely return `false`, no?

Reply:
Yes, this PR clearly still has some placeholder code.  I'll need to polish it some.